Update Microsoft.PolicyInsights .NET SDK

eladperets opened the pull request:
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec 
 changes has been included above.
Swagger pull requests:
https://github.com/Azure/azure-rest-api-specs/pull/16250
https://github.com/Azure/azure-rest-api-specs/pull/12661
https://github.com/Azure/azure-rest-api-specs/pull/10798
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

eladperets committed:
Vanilla commit

This change includes the output of generate.ps1

eladperets committed:
Revert unintentional generation changes

All tests pass (except remediations, since they need to be updated)

eladperets committed:
Update remediation tests

pilor committed:
CheckPolicyRestrictions tests

pilor committed:
Attestations tests

eladperets committed:
Update version

Yao725 reviewed (approved):
LGTM

Yao725 commented:
/azp run net - mgmt - ci

Yao725 commented:
/azp run net - mgmt - ci

eladperets commented:
@Yao725 Can you please merge this PR? Thanks!

Yao725 commented:
> @Yao725 Can you please merge this PR? Thanks!

Hi @eladperets , the ci failed because of the code discrepancy between your PR and reference generation. Have you modified the generated code in Generated folder and why?

eladperets commented:
> > @Yao725 Can you please merge this PR? Thanks!
> 
> Hi @eladperets , the ci failed because of the code discrepancy between your PR and reference generation. Have you modified the generated code in Generated folder and why?

@Yao725  yes, I have modified it. The generated code has methods with weird names like NextLink**1**WithHttpMessagesAsync, which we renamed to something more meaningful. This already existed in previous versions of the SDK, but every time we re-generate it we need to manually rename these methods.

I tried to break down the PR to several commits to  make it easier to see. The first commit contains the raw output of autorest. You can see it changes some of the method names. The second commit reverts the method names to their old and more readable names. The rest of the changes include the new changes to the API and their tests.

Yao725 commented:
> > > @Yao725 Can you please merge this PR? Thanks!
> > 
> > 
> > Hi @eladperets , the ci failed because of the code discrepancy between your PR and reference generation. Have you modified the generated code in Generated folder and why?
> 
> @Yao725 yes, I have modified it. The generated code has methods with weird names like NextLink**1**WithHttpMessagesAsync, which we renamed to something more meaningful. This already existed in previous versions of the SDK, but every time we re-generate it we need to manually rename these methods.
> 
> I tried to break down the PR to several commits to make it easier to see. The first commit contains the raw output of autorest. You can see it changes some of the method names. The second commit reverts the method names to their old and more readable names. The rest of the changes include the new changes to the API and their tests.

@eladperets Yeah, that makes sense to me and I can merge it. But I think it is a more efficient way to implement renaming by using directive which could guarantee CI pass and avoid manually changing it each time. Maybe we can do it in track 2 SDK.

Yao725 commented:
/check-enforcer override
